Z Score; how many standard deviations away from the mean 

Is the new bus route improving commute? Know what the travel time used to be mu=36, sigma=6



Alternative: H1: The change in the route for bus X8 reduced the travel time at peak hours. H1: old_mu > new_mu


NULL: H0: THe travel time for bus X8 is equal to or longer than the travel time before the route was changed. H0: old_mu <= new_mu

Significance level: alpha = .05
u1=36, std=6




Importing a bunch of tools 

In [21]:
from __future__ import division, print_function
import sys
import os
import json
import sys
import csv
import pandas as pd
import numpy as np
import pylab as py
from pandas import Series, DataFrame
%pylab inline

# add the urllib2 to get data from a url
import urllib2

Populating the interactive namespace from numpy and matplotlib


In [32]:
#url = 'https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt'
    
#response = urllib2.urlopen(url)
#data = response.read()
#print(data)
dataframe = pd.read_csv('https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt', header=None)
dataframe.head()

0
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [35]:
dataframe.mean()
dataframe.std()

0    7.13728
dtype: float64

In [51]:
old_mu = 36
old_sigma = 6
new_mu = dataframe.mean()
new_sigma = dataframe.std()
print('The new mean is ', new_mu)
print('The new std is', new_sigma)
N = len(dataframe) 
print(N)

The new mean is  0    34.466162
dtype: float64
The new std is 0    7.13728
dtype: float64
100


In [53]:
Z = (old_mu - new_mu)/(old_sigma/np.sqrt(N))
print(Z)

0    2.556397
dtype: float64


The Z score of 2.556 provides a p-value of .005. This value is outside of the .05 confidence level we initially set meaning that the change in the bus route lead to a statitically significant reduction in commute time. The Alternative hypothesis is proven. 

